### PR TITLE
Add email/pw auth, incl Amplitude + LogRocket

### PR DESCRIPTION
**Pull request by jwayne**
Demo: https://github.com/RomanEngeler1805/LegalResearchAssistant/assets/1303177/6b88a91a-d587-4202-9324-4958be536457

### CODE REVIEW

**Comment by RomanEngeler1805** [comment]
Looks good and works smooth to sign-up and login! A few points:

- I would suggest to implement an email / password recovery option if that's possible. I think it's a key feature to a user management system
- Is the choice deliberate to have sign-up and login on the same page? a couple of months ago, I did a tutorial about building a [study app](https://github.com/divanov11/StudyBud/blob/master/base/views.py), and they also had a login and sign-up functionality, might be useful as reference
- for the "Login" button color, adhere to the color palette specified in the [designs](https://www.figma.com/file/ws6r1jr3u3j1Xrz37hTtp9/Legal-Search?node-id=1456%3A61022&mode=dev) i.e the grey seen in the screenshot below
- how hard would it be to show the email next to the user icon as specified in the design? If it's a minimal effort (<30 mins), I would suggest doing it. Additionally, if it's minimal effort, also ask the user for the name in the sign-up process, such that we can personalise the experience by displaying their name as well
image

For a later stage, we want to design the login page as in our design
image

File: frontend/src/auth/Login.tsx [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        if (signInMethods.length) {
+          // The email already exists in the Auth database.
+          signInWithEmailAndPassword(auth, email, password)
+            .then(() => {
+              handleLoginSuccess()
+              setIsLoading(false)
+            })
+            .catch((error) => {
+              setError(
+                'Es ist ein Fehler aufgetreten. Bitte versuche es erneut.',
```
Comment (RomanEngeler1805) [inline frontend/src/auth/Login.tsx:53]: if the error is that the email is in the DB but the password is wrong, can we provide a more helpful error message? i.e. "Die Email Adresse und das Passwort stimmen nicht überein"

File: frontend/src/auth/LoginEmailLinkComplete.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      // Additional state parameters can also be passed via URL.
+      // This can be used to continue the user's intended action before triggering
+      // the sign-in operation.
+      // Get the email if available. This should be available if the user completes
+      // the flow on the same device where they started it.
+      let email = window.localStorage.getItem('emailForSignIn')
+      if (!email) {
+        // User opened the link on a different device. To prevent session fixation
+        // attacks, ask the user to provide the associated email again. For example:
+        email = window.prompt('Please provide your email for confirmation')
```
Comment (RomanEngeler1805) [inline frontend/src/auth/LoginEmailLinkComplete.tsx:23]: if user facing: "Bitte geb deine Email Adresse zur Bestätigung an"

File: frontend/src/auth/LoginEmailLinkComplete.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        if (!email) {
+          // eslint-disable-next-line no-alert
+          window.alert('You did not provide an email address')
```
Comment (RomanEngeler1805) [inline frontend/src/auth/LoginEmailLinkComplete.tsx:26]: if user facing: "Keine Email Adresse eingegeben"

**Comment by jwayne** [comment]
Hi Roman, I added the following:
- Password reset form
- Separate register form, with input box for name (note: you'll have to re-create the accounts you tested)
- Updated design according to the Figma (note: i didn't include the mosaic, since I wasn't sure what to do if the screen became too small to fit the mosaic)
- Show user name in menu

**Comment by RomanEngeler1805** [comment]
Looks neat! Tested it and works as expected.
I will merge it

**Pull request merged**